pallets/roles/src/benchmarking.rs

```diff
@@ -0,0 +1,144 @@
+use subsocial_support::{Content, User};
+
+fn create_dummy_space<T: Config + pallet_spaces::Config>(
+    origin: RawOrigin<T::AccountId>,
+) -> Result<Space<T>, DispatchError> {
+    let space_id = pallet_spaces::NextSpaceId::<T>::get();
+
+    pallet_spaces::Pallet::<T>::create_space(origin.into(), Content::None, None)
+        .map_err(|e| e.error)?;
+
+    let space = pallet_spaces::SpaceById::<T>::get(space_id)
+        .ok_or(DispatchError::Other("Space not found"))?;
+
+    Ok(space)
+}
+
+fn dummy_list_of_users<T: Config>(num_of_users: u32) -> Vec<User<T::AccountId>> {
+    let mut users_to_grant = Vec::<User<T::AccountId>>::new();
+
+    for i in 1..num_of_users + 1 {
+        let user = account("user", i * 2 - 1, i * 2);
+        users_to_grant.push(User::Account(user));
+    }
+
+    users_to_grant
+}
+
+fn create_dummy_role<T: Config>(
+    origin: RawOrigin<T::AccountId>,
+    space_id: SpaceId,
+    num_of_users: u32,
+) -> Result<(Role<T>, Vec<User<T::AccountId>>), DispatchError> {
+    let role_id = NextRoleId::<T>::get();
+
+    Pallet::<T>::create_role(
+        origin.clone().into(),
+        space_id,
+        Some(100u32.into()),
+        Content::None,
+        vec![SP::ManageRoles],
+    )?;
+
+    let role = RoleById::<T>::get(role_id).ok_or(DispatchError::Other("Role not found"))?;
+
+    let users_to_grant = dummy_list_of_users::<T>(num_of_users);
+
+    if !users_to_grant.is_empty() {
+        Pallet::<T>::grant_role(origin.into(), role.id, users_to_grant.clone())?;
+    }
+
+    Ok((role, users_to_grant))
+}
+
+benchmarks! {
+    where_clause { where T: pallet_spaces::Config }
+
+    create_role {
+        let caller_origin = RawOrigin::Signed(account::<T::AccountId>("Acc1", 1, 0));
+        let space = create_dummy_space::<T>(caller_origin.clone())?;
+        let time_to_live: Option<T::BlockNumber> = Some(100u32.into());
+        let content = Content::IPFS(b"QmRAQB6YaCyidP37UdDnjFY5vQuiBrcqdyoW1CuDgwxkD4".to_vec());
+        let perms = vec![SP::ManageRoles];
+        let role_id = NextRoleId::<T>::get();
+    }: _(caller_origin, space.id, time_to_live, content, perms)
+    verify {
+        let role = RoleById::<T>::get(role_id).unwrap();
+        let space_roles_ids = RoleIdsBySpaceId::<T>::get(space.id);
+
+        ensure!(role.id == role_id, "Role id doesn't match");
+        ensure!(space_roles_ids.contains(&role_id), "Role id not in space roles");
+    }
+
+    update_role {
+        let caller_origin = RawOrigin::Signed(account::<T::AccountId>("Acc1", 1, 0));
+        let space = create_dummy_space::<T>(caller_origin.clone())?;
+        let (role, _) = create_dummy_role::<T>(caller_origin.clone(), space.id, 10)?;
+
+        ensure!(!role.disabled, "Role should be enabled");
+
+        let update = RoleUpdate {
+             disabled: true.into(),
+             content: Content::IPFS(b"QmRAQB6YaCyidP37UdDnjFY5vQuiBrcqdyoW1CuDgwxkD4".to_vec()).into(),
```

Comment:
because of this:
```rs
    #[cfg(feature = "runtime-benchmarks")]
    type SpacePermissionsProvider = Spaces;
    #[cfg(not(feature = "runtime-benchmarks"))]
    type SpacePermissionsProvider = Self;
 ```
 
 SpacePermissionsProvider is implemented only for mocks that provide stubs that cannot be used for benchmarking.
 
 but all tests rely on these stubs, so when benchmarking are enabled I don't run tests